cmd/k8sruntime_util.go

```diff
@@ -93,17 +94,30 @@ func getContainers(resource k8sRuntime.Object) (container []Container) {
 	return container
 }
 
-func WriteToFile(decode k8sRuntime.Object, filename string) error {
+// WriteToFile writes and then appends incoming resource
+func WriteToFile(decode k8sRuntime.Object, filename string, toAppend bool) error {
 	info, _ := k8sRuntime.SerializerInfoForMediaType(scheme.Codecs.SupportedMediaTypes(), "application/yaml")
 	groupVersion := schema.GroupVersion{Group: decode.GetObjectKind().GroupVersionKind().Group, Version: decode.GetObjectKind().GroupVersionKind().Version}
 	encoder := scheme.Codecs.EncoderForVersion(info.Serializer, groupVersion)
 	yaml, err := k8sRuntime.Encode(encoder, decode)
 	if err != nil {
 		return err
 	}
-	err = ioutil.WriteFile(filename, yaml, 0644)
-	if err != nil {
-		return err
+	if !toAppend {
+		err = ioutil.WriteFile(filename, yaml, 0644)
+		if err != nil {
+			return err
+		}
+	} else {
+		f, err := os.OpenFile(filename, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+		defer f.Close()
```

Comment:
The `defer f.Close()` call needs to be moved to after the `err != nil` block because otherwise you might try to close on a non existent `f`